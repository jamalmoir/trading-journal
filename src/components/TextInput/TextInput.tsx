import React, { ChangeEvent } from 'react';

import styles from './textInput.scss';

interface TextInputProps {
  type: 'text' | 'email' | 'password';
  placeholder: string;
  value: string;
  errors?: boolean;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}

export const TextInput = (props: TextInputProps) => {
  let errorClass = props.errors ? styles.errors : '';

  return (
    <input className={ styles.textInput + ' ' + errorClass }
           type={ props.type }
           placeholder={ props.placeholder }
           onChange={ props.onChange }>
    </input>
  )
}