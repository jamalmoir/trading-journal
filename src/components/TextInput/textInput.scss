.textInput {
  margin: .5em 0;
  padding: .8em 1em;
  border: none;
  background: #eeeeee;

  &.errors {
    background: lightcoral;
  }

  ::placeholder {
    color: #cccccc;

  }
}