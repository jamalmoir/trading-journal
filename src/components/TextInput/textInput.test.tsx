import * as React from 'react';
import { TextInput } from './TextInput';
import { shallow } from 'enzyme';


describe("<TextInput />", () => {
  it('matches snapshot', () => {
    const component = shallow(<TextInput type='text' placeholder='test' value='test' />);
    expect(component).toMatchSnapshot();
  });

  it('handles input change', () => {
    const onChange = jest.fn();
    const component = shallow(<TextInput type='text' placeholder='test' value='test' onChange={ onChange } />);

    expect(onChange).not.toBeCalled();

    // enter text
    component.find('input[type="text"]').simulate('change', { target: { value: 'Hello' } })
    expect(onChange).toBeCalled();
  });

  it('has an errors class if conditions are met', () => {
    const component = shallow(<TextInput type='text' placeholder='test' value='test' />);

    // not touched, no errors
    component.setProps({touched: false, errors: []})
    expect(component.hasClass('errors')).toEqual(false);

    // not touched, errors
    component.setProps({touched: false, errors: ['an error']})
    expect(component.hasClass('errors')).toEqual(false);

    // touched, no errors
    component.setProps({touched: true, errors: []})
    expect(component.hasClass('errors')).toEqual(false);

    // touched, errors
    component.setProps({touched: true, errors: ['an error']})
    expect(component.hasClass('errors')).toEqual(true);
  });
});