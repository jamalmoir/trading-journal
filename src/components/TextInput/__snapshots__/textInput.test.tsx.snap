// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  className="undefined text-input form-control "
  placeholder="test"
  type="text"
  value="test"
/>
`;
